SpatialGDK/Source/SpatialGDK/Public/Utils/InspectionColours.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Containers/UnrealString.h"
+#include "Math/Color.h"
+#include "Math/UnrealMathUtility.h"
+#include "Misc/Char.h"
+
+// Mimicking Inspector V2 colouring from platform/js/console/src/inspector-v2/styles/colors.ts
+
+namespace SpatialGDK
+{
+	namespace
+	{
+		const int32 MIN_HUE = 10;
+		const int32 MAX_HUE = 350;
+		const int32 MIN_SATURATION = 60;
+		const int32 MAX_SATURATION = 100;
+		const int32 MIN_LIGHTNESS = 25;
+		const int32 MAX_LIGHTNESS = 60;
+
+		double GenerateValueFromThresholds(int64 Hash, int32 Min, int32 Max)
+		{
+			return Hash % FMath::Abs(Max - Min) + Min;
+		}
+
+		FColor HLAtoRGB(double Hue, double Saturation, double Lightness)
+		{
+			// const[h, s, l] = hsl;
+			// Must be fractions of 1
+
+			const double c = (1 - FMath::Abs(2 * Lightness / 100 - 1)) * Saturation / 100;
+			const double x = c * (1 - FMath::Abs(FMath::Fmod((Hue / 60), 2) - 1));
+			const double m = Lightness / 100 - c / 2;
+
+			double r = 0;
+			double g = 0;
+			double b = 0;
+
+			if (0 <= Hue && Hue < 60) {
+				r = c;
+				g = x;
+				b = 0;
+			}
+			else if (60 <= Hue && Hue < 120) {
+				r = x;
+				g = c;
+				b = 0;
+			}
+			else if (120 <= Hue && Hue < 180) {
+				r = 0;
+				g = c;
+				b = x;
+			}
+			else if (180 <= Hue && Hue < 240) {
+				r = 0;
+				g = x;
+				b = c;
+			}
+			else if (240 <= Hue && Hue < 300) {
+				r = x;
+				g = 0;
+				b = c;
+			}
+			else if (300 <= Hue && Hue <= 360) {
+				r = c;
+				g = 0;
+				b = x;
+			}
+			r = (r + m) * 255;
+			g = (g + m) * 255;
+			b = (b + m) * 255;
+
+			return FColor{ static_cast<uint8>(r), static_cast<uint8>(g), static_cast<uint8>(b) };
+		}
+	}
+
+	// Argument expected in the form: UnrealWorker1a2s3d4f...
+	FColor GetColourForWorkerId(const FString& WorkerId)
+	{
+		// DJB Reverse Hash
+		const int32 StringLength = WorkerId.Len();
+		int64 Hash = 5381;
+		for (int32 i = StringLength - 1; i > 0; --i) {
+			// We're mimicking the Inspector logic which is in JS. In JavaScript,
```

Comment:
As I understand, this logic is the same as in JS right now. However, there are no guarantees that JS logic will not change? And if it changes, the UE4 Editor and Inspector colors will not match again?

Reply:
Yea, I'm considering how we could find a way to inform the inspector of this through a WKC or worker details proto

Reply:
What if we generate colors on UE4 side and share them with the Inspector somehow (or the other way around)?

Reply:
I think it should be part of the viewport settings (inspector.json in v1) - so it could either be defined in a workspace file or some user provided configuration file for the inspector that the gdk would generate. Either way, I think it's an FR for observability that could come later. The main issue is that you want this configuration to be static in order to keep it simple, but it's hard to come up with this mapping. If we assume no modification to entity state to include some sort of well-defined virtual worker ID, I suggest agreeing on the mapping to be based on Worker component connected_since_utc field which both Unreal and the inspector can see. The user (Unreal) then needs to provide as configuration to the inspector a list of colours to use (the colours are assigned to workers in order of connection time).